pkg/serializer/jsonstream/compressor.go

```diff
@@ -44,9 +46,8 @@ func init() {
 
 // the backend accepts payloads up to 3MB/50MB, but being conservative is okay
 var (
-	megaByte            = 1024 * 1024
-	maxPayloadSize      = 2*megaByte + megaByte/2 // `2.5*megaByte` won't work with strong typing
-	maxUncompressedSize = 45 * megaByte
+	maxPayloadSize      = config.Datadog.GetInt("serializer_max_payload_size")
```

Comment:
Is the config guaranteed to be parsed this early that we can use it in a global assignment? This might just always return the default.

Reply:
You're right, moving this at the compressor creation